src/main/java/com/zepben/evolve/database/sqlite/writers/BaseCIMWriter.kt

```diff
@@ -49,6 +53,46 @@ abstract class BaseCIMWriter(protected val databaseTables: DatabaseTables) {
         return saveIdentifiedObject(table, insert, organisation, "organisation")
     }
 
+    fun save(nameType: NameType): Boolean {
+        val table = databaseTables.getTable(TableNameType::class.java)
+        val insert = databaseTables.getInsert(TableNameType::class.java)
+
+        return saveNameType(table, insert, nameType)
+    }
+
+    fun save(name: Name): Boolean {
+        val table = databaseTables.getTable(TableName::class.java)
+        val insert = databaseTables.getInsert(TableName::class.java)
+
+        return saveName(table, insert, name)
+    }
+
+    private fun saveNameType(table: TableNameType, insert: PreparedStatement, nameType: NameType): Boolean {
+        insert.setString(table.NAME.queryIndex, nameType.name)
+        insert.setNullableString(table.DESCRIPTION.queryIndex, nameType.description)
+
+        return tryExecuteSingleUpdate(insert, nameType.name, "name type")
+    }
+
+    private fun saveName(table: TableName, insert: PreparedStatement, name: Name): Boolean {
+        var status = true
+
+        if (!name.identifiedObjects.isEmpty()) {
+            name.identifiedObjects.forEach {
+                insert.setString(table.NAME.queryIndex, name.name)
+                insert.setNullableString(table.NAME_TYPE_NAME.queryIndex, name.type?.name)
```

Comment:
Your schema says this field and the next field are not null. I'm happy for this. I'm probably also happy to remove the else statement below and we just document names with no associated identified objects will not be written to the database.